core/LSU.cpp

```diff
@@ -482,16 +905,56 @@ namespace olympia
             }
         }
 
-        sparta_assert(false, "Attempt to complete instruction no longer exiting in issue queue!");
+        sparta_assert(false, "Attempt to complete instruction no longer exiting in issue queue! " << inst_ptr);
+    }
+
+    void inline LSU::appendToQueue_(LoadStoreIssueQueue &queue, const LoadStoreInstInfoPtr &inst_info_ptr){
+        sparta_assert(queue.size() < ldst_inst_queue_size_,
+                      "Appending load queue causes overflows!");
+
+        bool exists = std::find_if(
+            queue.begin(), queue.end(),
+            [inst_info_ptr](auto &queue_inst_ptr) { return queue_inst_ptr->getInstPtr() == inst_info_ptr->getInstPtr(); }
+        ) == std::end(queue);
+
+        sparta_assert(exists,  "Cannot push duplicate instructions into the queue " << inst_info_ptr);
+
+        // Always append newly dispatched instructions to the back of issue queue
+        queue.push_back(inst_info_ptr);
+
+        ILOG("Append new instruction to queue!");
     }
 
     // Arbitrate instruction issue from ldst_inst_queue
     const LSU::LoadStoreInstInfoPtr & LSU::arbitrateInstIssue_()
     {
         sparta_assert(ldst_inst_queue_.size() > 0, "Arbitration fails: issue is empty!");
 
-        // Initialization of winner
-        auto win_ptr_iter = ldst_inst_queue_.begin();
+        auto win_ptr_iter = replay_buffer_.begin();
+        if(allow_speculative_load_exec_)
+        {
+            for (auto iter = replay_buffer_.begin(); iter != replay_buffer_.end(); iter++)
+            {
+                if (!(*iter)->isReady())
+                {
+                    continue;
+                }
+                if (!(*win_ptr_iter)->isReady() || ((*iter)->winArb(*win_ptr_iter)))
+                {
+                    win_ptr_iter = iter;
+                }
+            }
+            if (win_ptr_iter.isValid() && (*win_ptr_iter)->isReady() && allow_speculative_load_exec_)
```

Comment:
Remove `allow_speculative_load_exec_` from the condition.  You've already checked for this.